eland/ml/pytorch/pytorch_model.py

```diff
@@ -0,0 +1,122 @@
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+import base64
+import json
+import math
+import os
+from eland.common import ensure_es_client
+from tqdm import tqdm
+from typing import TYPE_CHECKING, Any, Dict, List, Tuple, Union
+
+if TYPE_CHECKING:
+    from elasticsearch import Elasticsearch
+
+DEFAULT_CHUNK_SIZE = 4 * 1024 * 1024  # 4MB
+QUEUE_SIZE = 4
+THREAD_COUNT = 4
+
+
+class PyTorchModel:
+    """
+    A PyTorch model managed by Elasticsearch.
+
+    These models must be trained outside of Elasticsearch, conform to the
+    support tokenization and inference interfaces, and exported as their
+    TorchScript representations.
+    """
+
+    def __init__(
+        self,
+        es_client: Union[str, List[str], Tuple[str, ...], "Elasticsearch"],
+        model_id: str,
+    ):
+        self._client = ensure_es_client(es_client)
+        # Elasticsearch model IDs need to be a specific format: no special chars, all lowercase, max 64 chars
+        self.model_id = model_id.replace('/', '__').lower()[:64]
```

Comment:
Instead of silently trimming/cleaning up we should raise a `ValueError` with these requirements.

Reply:
Ok, this is an "automagic" conversion that we need to do, but you're right, it belongs somewhere else (in the place where the ID is automatically created from another model). 💯

Reply:
Validation is happening already at upload time so I'll leave it there for now